tests/test_plugin_loader.py

```
@@ -4463,5 +4591,67 @@ def test_update_existing_repo_to_branch_or_tag_fetch_failure(
+
+        # Clean up any leftover entries from previous runs
+        for name in module_names:
+            sys.modules.pop(name, None)
+
+        try:
+            with concurrent.futures.ThreadPoolExecutor(max_workers=4) as pool:
+                futures = [pool.submit(_load_module, name) for name in module_names]
+                concurrent.futures.wait(futures)
+
+            # After all threads complete, none of the test modules should linger
+            # unless the exec succeeded – in that case the module is expected.
```

Ruff (0.15.7)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)